spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java

```diff
@@ -219,13 +241,41 @@ public ConsulDiscoveryProperties(InetUtils inetUtils) {
 
 	/**
 	 * @param serviceId The service who's filtering tag is being looked up
-	 * @return The tag the given service id should be filtered by, or null.
+	 * @return The tags the given service id should be filtered by, or null.
+	 * @deprecated instead use {@link #getQueryTagsForService(String)}
 	 */
+	@Deprecated
 	public String getQueryTagForService(String serviceId) {
```

Comment:
While this method attempts to first use the service specific query tag, its not used by anything internally other than tests. I deprecated it in favor of the following plural version.